### PR TITLE
initial experimentations on bazel build

### PR DESCRIPTION
**Pull request by themarwhal**
Signed-off-by: Marie Bremner <marwhal@fb.com>

## Important
use this new dockerfile
```
cd lte/gateway/docker/mme
docker build -t bazel_base -f Dockerfile.ubuntu20.04.bazel ../../../../
docker run -v $MAGMA_ROOT:/magma -v $MAGMA_ROOT/lte/gateway/configs:/etc/magma -i -t bazel_base:latest /bin/bash
```

### Running Bazel commands
```
cd /magma
bazel build $PATH_TO_PACKAGE:$LIBRARY_NAME
bazel build ...
bazel test ...
# example: bazel build orc8r/gateway/c/common/config:config
```

### Notable Changes
1. upgraded cpp-redis to be latest release since that version included Bazel support. This meant I've had to slightly fix up some types but nothing too big.
2. session manager on master has two sub directories called datastore and policdb. It was annoying to deal with the dependency chain across directories, so I just flattened it into the main session manager directory. (The two sub directories in total had 5 files, so I figured it was no big deal)

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/session_manager/magma_logging.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#include <iostream>
+
+// for non glog use cases, just log to std cout
+struct _MLOG_NEWLINE {
+  ~_MLOG_NEWLINE() { std::cout << std::endl; }
+};
+#define MLOG(VERBOSITY)                                                        \
+  (_MLOG_NEWLINE(), std::cout << "[" << __FILE__ << ":" << __LINE__ << "] ")
+
+#endif
```
Comment (uri200) [inline lte/gateway/c/session_manager/magma_logging.h:41]: isnt this a common library?

or are you placing this just for test?

Comment (themarwhal) [reply lte/gateway/c/session_manager/magma_logging.h:41]: This I just placed until I figure out how to import it from common

Taking some shortcuts to get some sort of build first :)

Comment (themarwhal) [reply lte/gateway/c/session_manager/magma_logging.h:41]: Well importing isn't the hard part. Making it work for bazel while not breaking the cmake build is harder😩

Comment (uri200) [reply lte/gateway/c/session_manager/magma_logging.h:41]: well copying temporary there to allow a migration is fine. We can remove it later

Problem is if this happens with many more.

Is it possible to have hybrid builds, meaning, sessiond all using Bazel, but the rest still using cmake?

Comment (themarwhal) [reply lte/gateway/c/session_manager/magma_logging.h:41]: should be yes! they are separate builds already, so that should not be a problem

**Pull request closed without merging**